dartsim/src/JointFeatures.cc

```diff
@@ -156,7 +156,56 @@ void JointFeatures::DetachJoint(const Identity &_jointId)
       child->getSpatialVelocity(
           dart::dynamics::Frame::World(),
           dart::dynamics::Frame::World());
-  auto freeJoint = child->moveTo<dart::dynamics::FreeJoint>(nullptr);
+
+  if (!this->links.HasEntity(child))
+  {
+    return;
+  }
+
+  auto childLinkInfo = this->links.at(child);
+
+  dart::dynamics::SkeletonPtr skeleton;
+  {
+    // Find the original skeleton the child BodyNode belonged to
+    std::string oldName = child->getName();
+    if (oldName != childLinkInfo->name)
+    {
+      std::size_t originalNameIndex = oldName.find(childLinkInfo->name);
```

Comment:
I think we should use an `rfind` so we don't accidentally find the skeleton's name?

Reply:
Good point. I added a failing test that shows what happens when a model and link have the same name in 53314df and applied your suggestion in 7a9e269, which fixes the test.